src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -88,48 +86,62 @@ internal void LogDiagnostic(Diagnostic diagnostic, CultureInfo culture)
 
         private void WriteLocations(Location location, IReadOnlyList<Location> additionalLocations)
         {
-            _writer.WriteArrayStart("locations");
+            if (location.SourceTree != null)
+            {
+                _writer.WriteArrayStart("locations");
+                _writer.WriteObjectStart(); // location
+                _writer.WriteKey("analysisTarget");
```

Comment:
I think that Nick's observation -- that the compiler layer doesn't even have the concept of a "project file" -- disposes of the suggestion that the project file be the analysisTarget.

That leaves Michael's question of what happens when SharedFile.cs is compiled into multiple projects. The answer is as follows:

As MSBuild builds the solution, it will kick off a separate compilation for each project that uses SharedFile.cs:

csc a.cs ... SharedFile.cs ... /out:foo.dll /errorlog:foo.sarif.json

csc b.cs ... SharedFile.cs ... /out:bar.dll /errorlog:bar.sarif.json

So diagnostics in SharedFile.cs will appear in two separate log files. Each log file will have a run.invocation.commandLine that says what was being built.

---
In reply to: [63097985](https://github.com/dotnet/roslyn/pull/11233#discussion_r63097985) [](ancestors = 63097985)